### PR TITLE
[Data] Add release test for TPCH Q1

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

To make sure Ray Data can performantly handle complex data processing workloads, this PR adds a TPCH benchmark to our release tests.
 
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/dataset/tpch_q1.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="TPCH Q1")
+    parser.add_argument("--sf", choices=[1, 10, 100, 1000, 10000], type=int, default=1)
+    return parser.parse_args()
+
+
+def main(args):
+    path = f"s3://ray-benchmark-data/tpch/parquet/sf{args.sf}/lineitem"
+    benchmark = Benchmark("tpch-q1")
+
```
Comment (srinathk10) [inline release/nightly_tests/dataset/tpch_q1.py:23]: Nice!

To work around the column names annoyance, we can add rename_columns after read_parquet or just use a Dict of the current Dataset column names to TPCH Lineitem column names. This should improve the code readability significantly.

File: release/nightly_tests/dataset/tpch_q1.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                Sum(on="column05", alias_name="sum_base_price"),  # l_extendedprice
+                Sum(on="disc_price", alias_name="sum_disc_price"),
+                Sum(on="charge", alias_name="sum_charge"),
+                Mean(on="column04", alias_name="avg_qty"),  # l_quantity
+                Mean(on="column05", alias_name="avg_price"),  # l_extendedprice
+                Mean(on="column06", alias_name="avg_disc"),  # l_discount
+                Count(),  # FIXME: No way to specify column name
+            )
+            .sort(["column08", "column09"])  # l_returnflag, l_linestatus
+            .materialize()
```
Comment (omatthew98) [inline release/nightly_tests/dataset/tpch_q1.py:46]: Do these aggregations force materialization or why do we need to explicitly materialize here?

File: release/nightly_tests/dataset/tpch_q1.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    return parser.parse_args()
+
+
+def main(args):
+    path = f"s3://ray-benchmark-data/tpch/parquet/sf{args.sf}/lineitem"
+    benchmark = Benchmark("tpch-q1")
+
+    def benchmark_fn():
+        (
+            ray.data.read_parquet(path)
```
Comment (omatthew98) [inline release/nightly_tests/dataset/tpch_q1.py:26]: Maybe add a quick comment on what we are testing for here, seems like it is just filtering down a particular slice of the data then testing aggregates on multiple columns but might be good to add why we are doing multiple columns etc.

**Review by omatthew98 — APPROVED** [review]
2 questions but lgtm.

File: release/nightly_tests/dataset/tpch_q1.py (hunk 2 above)
Comment (bveeramani) [reply release/nightly_tests/dataset/tpch_q1.py:46]: The aggregations force materialization, but they don't trigger execution. So, we need `materialize` to actually execute the pipeline (even though the data is already materialized)

File: release/nightly_tests/dataset/tpch_q1.py (hunk 3 above)
Comment (bveeramani) [reply release/nightly_tests/dataset/tpch_q1.py:26]: Ah, good call. I just directly adapted the TPC-H query code from https://examples.citusdata.com/tpch_queries.html.

Added a comment

**Pull request merged**